src/Order/Checkout/CheckoutForm.php

```diff
@@ -0,0 +1,269 @@
+     * @throws \SilverStripe\Omnipay\Exception\InvalidStateException
+     * @throws \Exception
+     */
+    public function ConfirmCheckout(array $data): HTTPResponse
+    {
+        $this->cart->Lock();
+        $this->extend('beforeInitPayment', $data);
+
+        $this->saveInto($this->cart);
+        $this->cart->write();
+
+        /** @var Payment|PaymentExtension $payment */
+        $payment = Payment::create();
+        $payment->OrderID = $this->cart->ID;
+
+        $paymentMethod = $data[static::PAYMENT_METHOD_FIELD];
+        $dueMoney = $this->cart->UnpaidTotal()->getMoney();
+        $payment->init($paymentMethod, $this->supportedCurrencies->formatDecimal($dueMoney),
+            $dueMoney->getCurrency()->getCode())
+            ->setSuccessUrl($this->getSuccessUrl())
+            ->setFailureUrl($this->getFailureUrl($payment));
+
+        if ($payment->isChanged(null, Payment::CHANGE_VALUE)) {
+            $payment->write();
+        }
+
+        $response = $this->paymentServiceFactory
+            ->getService($payment, ServiceFactory::INTENT_PURCHASE)
+            ->initiate($data);
+
+        $this->extend('afterInitPayment', $data, $payment, $response);
+        return $response->redirectOrRespond();
+    }
+
+    /**
+     * @return string
+     */
+    protected function getSuccessUrl(): string
+    {
+        /** @var OrderConfirmationPage $orderConfirmationPage */
+        $orderConfirmationPage = OrderConfirmationPage::get_one(OrderConfirmationPage::class);
+        return $orderConfirmationPage->LinkForOrder($this->cart);
+    }
+
+    /**
+     * @param Payment $payment
+     * @return string
+     * @throws \SilverStripe\ORM\ValidationException
+     */
+    protected function getFailureUrl(Payment $payment): string
+    {
+        if (!$payment->Identifier) {
+            // Force identifier to be generated
+            /** @see Payment::onBeforeWrite() */
+            $payment->write();
+        }
+
+        return Controller::join_links(
+            $this->getController()->Link(),
+            sprintf('?%1$s=%2$s', static::PAYMENT_ID_QUERY_PARAM, $payment->Identifier)
+        );
+    }
+
+    /**
+     * @inheritdoc
+     * @throws \SilverStripe\Omnipay\Exception\InvalidConfigurationException
+     */
+    protected function buildFields(): FieldList
+    {
+        $fields = FieldList::create([
+            TextField::create('CustomerName', 'Name'),
+            EmailField::create('CustomerEmail', 'Email'),
+            $this->cart->BillingAddress->scaffoldFormField('Billing Address'),
+        ]);
+
+        $gateways = GatewayInfo::getSupportedGateways();
+        $gatewayField = count($gateways) > 1
+            ? OptionsetField::create(static::PAYMENT_METHOD_FIELD,
+                _t(self::class . '.PAYMENT_METHOD', 'Select your payment method'), $gateways)
+            : HiddenField::create(static::PAYMENT_METHOD_FIELD, null, key($gateways));
```

Comment:
If you have only a single allowed gateway, using a hidden field requires you to validate it. It's better to not have a field at all, since the value becomes a security attack vector.

In your form submission logic you simply check if there are multiple supported options, and only check the submitted value if there are multiple.